akka-stream/src/main/scala/akka/stream/scaladsl/FlowWithContextOps.scala

```diff
@@ -43,6 +44,21 @@ trait FlowWithContextOps[+Out, +Ctx, +Mat] {
    */
   def via[Out2, Ctx2, Mat2](flow: Graph[FlowShape[(Out, Ctx), (Out2, Ctx2)], Mat2]): Repr[Out2, Ctx2]
 
+  /**
+   * Transform this flow by the regular flow. The given flow works on the data portion of the stream and
+   * ignores the context.
+   *
+   * The given flow *must* not re-order, drop or emit multiple elements for one incoming
+   * element, the sequence of incoming contexts is re-combined with the outgoing
+   * elements of the stream. If a flow not fulfilling this requirement is used the stream
+   * will not fail but continue running in a corrupt state and re-combine incorrect pairs
+   * of elements and contexts.
```

Comment:
With broadcast, if the element is dropped I think it will deadlock (broadcast only emits once there is demand from both downstreams, but zip wont request another element until it got one down each upstream), so for completeness:

```suggestion
   * of elements and contexts or deadlock.
```

Reply:
Committed and pushed.